okio/src/nonJvmMain/kotlin/okio/ForwardingSource.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+/** A [Source] which forwards calls to another. Useful for subclassing. */
+actual abstract class ForwardingSource actual constructor(
+  /** [Source] to which this instance is delegating. */
+  actual val delegate: Source
+) : Source {
+  // TODO 'Source by delegate' once https://youtrack.jetbrains.com/issue/KT-23935 is fixed.
+
+  actual override fun read(sink: Buffer, byteCount: Long): Long = delegate.read(sink, byteCount)
```

Comment:
Curious about exception standards, should these be marked here?

Reply:
I think it's only needed if this wasn't set already in the expect class. I would look at other actuals, like FileSystem to compare.

Reply:
Following FileSystem I put them in.